### PR TITLE
Added a solution to the challenge 'hobbyTracker' in CSX/Closures

### PR DESCRIPTION
**Pull request by a-hend**
Cole - if you have time to review this one, I'd love to hear what you have to say.  I'm pretty certain the code could be much cleaner.

### CODE REVIEW

File: Users/abehenderson/CSX/Closures/hobbyTracker.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  return function newFunc(hobby, hours) {
+    if (hobby === undefined && hours === undefined) {
+      for (const keys in cache) {
+        cache[keys] = 0;
+      }
+      return 'tracker has been reset!'
+    } else {
+      for (const property in cache) {
+         if (hobby === property) cache[property] += hours;
+      }
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Closures/hobbyTracker.js:17]:
```suggestion
     cache[hobby] += hours; 
      }
```

File: Users/abehenderson/CSX/Closures/hobbyTracker.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    return cache;
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Closures/hobbyTracker.js:19]: you only want to return the cache if hobby and hours are defined (see directions).

**Review by C-STYR — APPROVED** [review]
Your code is fine, I suggested one refinement and one clarification.

File: Users/abehenderson/CSX/Closures/hobbyTracker.js (hunk 2 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Closures/hobbyTracker.js:19]: This make sense, thanks

File: Users/abehenderson/CSX/Closures/hobbyTracker.js (hunk 1 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Closures/hobbyTracker.js:17]: I'm not sure I understand how that would work.  `property` is declared in the `for...in` loop.  How else would I examine all of the keys to see if they match `hobby`?

Comment (C-STYR) [reply Users/abehenderson/CSX/Closures/hobbyTracker.js:17]: That's the point of a JS object: you don't have to. I've edited my suggestion above to remove the comment so it's clearer. To access a property on an object, just call it directly: `cache[hobby] += hours`.  No need to loop.

**Pull request merged**